### PR TITLE
 Addressing issue bincrafters/community#140

### PR DESCRIPTION
**Pull request by ghost**
Modifications have been tested locally since the test_project CMakeLists file had to be temporarily changed to use cmake find modules instead of the CONAN_LIBS variable.

Some assumptions have been made, especially in the way FindGMock works. Please let me know if a different approach is more desirable.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -11,45 +11,53 @@ class GTestConan(ConanFile):
     description = "Google's C++ test framework"
     url = "http://github.com/bincrafters/conan-gtest"
     license = "BSD 3-Clause"
-    exports = ["LICENSE.md"]
-    exports_sources = ["CMakeLists.txt"]
+    exports = ["LICENSE.md", "FindGTest.cmake", "FindGMock.cmake"]
```
Comment (solvingj) [inline conanfile.py]: Should move these to export_sources

File: test_package/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,10 @@ conan_basic_setup()
 
 file(GLOB SOURCE_FILES *.cpp)
 
+if(WITH_GMOCK)
+    add_definitions("-DWITH_GMOCK")
```
Comment (solvingj) [inline test_package/CMakeLists.txt:12]: We should probably conditionally declare this from `package_info` int he package no?

File: test_package/conanfile.py [hunk 3]
```diff
@@ -11,6 +11,7 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        cmake.definitions['WITH_GMOCK'] = self.options['gtest'].build_gmock
```
Comment (solvingj) [inline test_package/conanfile.py:14]: If we declare this from the main package, we won't need to do this.

**Review by solvingj — COMMENTED** [review]
This is great work!  We really needed this, thank you!

File: conanfile.py (hunk 1 above)
Comment (ghost) [reply conanfile.py]: Indeed! Done in the latest commit (just now).

File: test_package/CMakeLists.txt (hunk 2 above)
Comment (ghost) [reply test_package/CMakeLists.txt:12]: Normally that would be correct, but this is not a gmock/gtest manipulation flag. It has been specifically created for the test of this conan package from what I see (I did not create it myself). It is only used in test_package/test_package.cpp to activate/deactivate the part of the code that tests gmock, depending on whether the package has been built with gmock or not.

File: test_package/conanfile.py (hunk 3 above)
Comment (ghost) [reply test_package/conanfile.py:14]: (my comment above applies here too)

**Pull request merged**